lib/matrix_sdk/request.ex

```diff
@@ -293,12 +295,12 @@ defmodule MatrixSDK.Request do
         path: "/_matrix/client/r0/register"
       }
   """
-  @spec register_user(base_url, binary, map) :: t
-  def register_user(base_url, password, opts \\ %{}) do
+  @spec register_user(base_url, binary, Auth.t(), map) :: t
+  def register_user(base_url, password, auth, opts \\ %{}) do
     body =
```

Comment:
```suggestion
    body = %{password: password, auth: auth} |> Map.merge(opts)
```

Not a big deal, but it is possible to turn this into a one liner

Reply:
I generally try to avoid piping once as a style rule and so the multiple pipes are consistent with the rest of the code.
We could probably refactor a lot of those in future to something like `Map.merge(%{password: password, auth: auth}, opts)` 👍